### PR TITLE
Di 22066 dynamic getter

### PR DESCRIPTION
**Pull request by roiaharon-pepperi**

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -119,4 +119,18 @@ export function filter<T>(objects: T[], jsonFilter: JSONFilter | undefined): T[]
...
+        const filterFactory = new FilterFactory();
+        const filter = filterFactory.createFilter(jsonFilter);
+        objects = await Filter.filterAsync(objects, filter);
+    }
+    return objects.map((obj) => obj.Item);
+}
+
+export interface IFilterObject<T> {
+    Item: T;
+    GetValue: (apiName: string) => Promise<any>;
```
Comment (Chasky-H) [inline index.ts:133]: should be lowercase.

Comment (Chasky-H) [reply index.ts:133]: camelCase

Comment (Chasky-H) [reply index.ts:133]: fieldID

Comment (roiaharon-pepperi) [reply index.ts]: done

**Pull request merged**